.github/workflows/pyrealm_ci.yaml

```diff
@@ -55,22 +55,36 @@ jobs:
       env:
         CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
 
-    - name: Run profiling
+    - name: Run the profiling test suite
       id: profiling
-      if: steps.lfs.outcome == 'success'
+      if: success() && (matrix.os == 'ubuntu-latest' && matrix.python-version == 3.10)
       run: |
         sudo apt-get install graphviz
         poetry run pytest -m "profiling" --profile-svg
-
-    - name: Report performance (this step will fail if performance degrades)
-      if: steps.profiling.outcome == 'success'
-      run: poetry run python profiling/report.py --event ${{ github.event_name }}
-
+        calculatedSha=$(git rev-parse --short ${{ github.sha }})
+        echo "COMMIT_SHORT_SHA=$calculatedSha" >> $GITHUB_ENV
+
+    # This step only runs if the profiling succeeds - so only runs on ubuntu/3.10 and 
+    # will itself fail if the benchmarking fails
+    - name: Benchmark performance
+      id: benchmarking
+      if: ${{steps.profiling.outcome == 'success'}}
+      run: poetry run python profiling/run_benchmarking.py prof/combined.prof profiling/profiling-database.csv profiling/benchmark-fails.csv ${{ env.COMMIT_SHORT_SHA }} --update-on-pass --plot-path profiling/performance-plot.png
+
+    # If the benchmarking succeeds, then update the call graph. This again only runs on
+    # the ubuntu/3.10 runner used for the profiling.
+    - name: Update call graph
+      id: update_call_graph
+      if: ${{steps.benchmarking.outcome == 'success'}}
+      run: cp prof/combined.svg profiling/call-graph.svg
```

Comment:
Maybe this command can be appended as a subsequent command in the last step (Benchmark performance)? If the `poetry run ...` command fails, then this command will be automatically skipped.

Reply:
The intention was to deliberately only update the call graph on passing benchmark tests - but I'm not sure that's necessarily what we want, thinking about it?

I guess my reasoning is that the call graph shows the profiling of a single version of the code - and we'd like the website to show only the profile of the current release that passes everything. But only passing versions are going to get merged, so we could just copy every time.

Reply:
Yeah - on reflection, I think that is better.